pkg/secrets/config/config.go

```diff
@@ -49,10 +54,22 @@ func NewFromEnv() (*Config, error) {
 		return nil, err
 	}
 
+	timeout := DEFAULT_TIMEOUT
+	if envTimeout, _ := strconv.Atoi(os.Getenv("TIMEOUT")); envTimeout != DEFAULT_TIMEOUT && envTimeout != 0 {
+		timeout = envTimeout
+	}
+
+	retryCountLimit := DEFAULT_RETRY_COUNT_LIMIT
+	if envRetryCountLimit, _ := strconv.Atoi(os.Getenv("RETRY_COUNT_LIMIT")); envRetryCountLimit != DEFAULT_RETRY_COUNT_LIMIT && envRetryCountLimit != 0 {
+		retryCountLimit = envRetryCountLimit
+	}
```

Comment:
You can extract this method into something like:
```
ParseIntFromEnvOrDefault("RETRY_COUNT_LIMIT", DEFAULT_RETRY_COUNT_LIMIT)
```
Then you can reuse it for both env vars.